### PR TITLE
feat:Add rest call support similar to oepn-api style

### PR DESCRIPTION
**Pull request by wuhongsheng**

### CODE REVIEW

**Comment by andimarafioti** [comment]
You're such a goat! Thank you 🙏 
With which base_url were you testing this? for me to reproduce

**Comment by wuhongsheng** [comment]
> You're such a goat! Thank you 🙏 With which base_url were you testing this? for me to reproduce

I tested it with deepseek url

from openai import OpenAI

client = OpenAI(api_key="<DeepSeek API Key>", base_url="https://api.deepseek.com")

response = client.chat.completions.create(
    model="deepseek-chat",
    messages=[
        {"role": "system", "content": "You are a helpful assistant"},
        {"role": "user", "content": "Hello"},
    ],
    stream=False
)

print(response.choices[0].message.content)

**Comment by MattiPaivike** [comment]
Yes, for this to be useful it should stream the text from the API and create TTS for each chunked text stream. Too much lag if you have to wait for generation of the entire message.

There is an example of how streaming can be implemented in OpenAI Python library:
https://github.com/openai/openai-python?tab=readme-ov-file#streaming-responses

File: arguments_classes/open_api_language_model_arguments.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    )
+    open_api_init_chat_role: str = field(
+        default="system",
+        metadata={
+            "help": "Initial role for setting up the chat context. Default is 'system'."
+        },
+    )
+    open_api_init_chat_prompt: str = field(
+        # default="You are a helpful and friendly AI assistant. You are polite, respectful, and aim to provide concise responses of less than 20 words.",
+        default="你是一位乐于助人且友好的 AI 助手。您彬彬有礼、尊重他人.",
```
Comment (eustlb) [inline arguments_classes/open_api_language_model_arguments.py:27]: Let's stick with a default initial chat prompt in english. 

```suggestion
        default="You are a helpful and friendly AI assistant. You are polite, respectful, and aim to provide concise responses of less than 20 words.",
```

File: LLM/openai_api_language_model.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        response = self.client.chat.completions.create(
+            model=self.model_name,
+            messages=[
+                {"role": self.user_role, "content": prompt},
+            ],
+            stream=self.stream
+        )
+        generated_text = response.choices[0].message.content
+        self.chat.append({"role": "assistant", "content": generated_text})
+        yield generated_text
```
Comment (eustlb) [inline LLM/openai_api_language_model.py:71]: Let's handle streaming as done in [language_model.py](https://github.com/huggingface/speech-to-speech/blob/d98e25276976c16e6881b01dc4dfb625021e9291/LLM/language_model.py#L113C5-L144C46)

```suggestion
def process(self, prompt):
        logger.debug("call api language model...")
        self.chat.append({"role": self.user_role, "content": prompt})

        language_code = None
        if isinstance(prompt, tuple):
            prompt, language_code = prompt

        response = self.client.chat.completions.create(
            model=self.model_name,
            messages=[
                {"role": self.user_role, "content": prompt},
            ],
            stream=self.stream
        )
        if self.stream:
            generated_text, printable_text = "", ""
            for chunk in response:
                new_text = chunk.choices[0].delta.content or ""
                generated_text += new_text
                printable_text += new_text
                sentences = sent_tokenize(printable_text)
                if len(sentences) > 1:
                    yield sentences[0], language_code
                    printable_text = new_text
            self.chat.append({"role": "assistant", "content": generated_text})
            # don't forget last sentence
            yield printable_text, language_code
        else:
            generated_text = response.choices[0].message.content
            self.chat.append({"role": "assistant", "content": generated_text})
            yield generated_text, language_code
```

**Review by eustlb — CHANGES_REQUESTED** [review]
Great work, thanks @wuhongsheng ! 🙏🤗
This will definitely meet a relevant use case.
Almost ready to merge, it just needs to handle the `--open_api_stream` flag. The suggestion given below to handle it has been tested using `--whisper_mlx` and `--melo`, works fine.

File: LLM/openai_api_language_model.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
+from openai import OpenAI
+from LLM.chat import Chat
+from baseHandler import BaseHandler
+from rich.console import Console
+import logging
+import time
+logger = logging.getLogger(__name__)
+
+console = Console()
+
```
Comment (wuhongsheng) [inline LLM/openai_api_language_model.py:10]: from nltk import sent_tokenize

**Review by wuhongsheng — COMMENTED** [review]
add sent_tokenize

**Review by wuhongsheng — COMMENTED** [review]
is OK

**Comment by eustlb** [comment]
Hey @wuhongsheng, thanks for iterating. Nevertheless, you have not committed [suggested changes](https://github.com/huggingface/speech-to-speech/pull/81#discussion_r1747333293) to handle the `--open_api_stream` flag, can you do so ? After that, we'll be able to merge 👌

**Review by wuhongsheng — COMMENTED** [review]
is ok

**Comment by eustlb** [comment]
LGTM! Thanks for the good work 🤗

**Pull request merged**